Formula/pdns.rb

```diff
@@ -26,54 +27,79 @@ class Pdns < Formula
   deprecated_option "with-pgsql" => "with-postgresql"
 
   depends_on "pkg-config" => :build
+  depends_on "cmake" => :build
   depends_on "boost"
   depends_on "lua"
   depends_on "openssl"
   depends_on "sqlite"
   depends_on :postgresql => :optional
 
+  resource "pdns-brew-default-config" do
+    url "https://github.com/mprzybylski/pdns-brew-default-config/archive/0.2.tar.gz"
+    sha256 "5abe291cbd789688504731e08cf098a8b5afda9f1928c20c4c0ba8eef025e6b5"
+  end
   def install
     args = %W[
       --prefix=#{prefix}
+      --sysconfdir=#{etc}/pdns
+      --localstatedir=#{var}
+      --with-socketdir=#{var}/run
       --with-lua
       --with-openssl=#{Formula["openssl"].opt_prefix}
       --with-sqlite3
     ]
 
+    # default backend modules
+    module_list="--with-modules=bind gsqlite3"
+
     # Include the PostgreSQL backend if requested
-    if build.with? "postgresql"
-      args << "--with-modules=gsqlite3 gpgsql"
-    else
-      # SQLite3 backend only is the default
-      args << "--with-modules=gsqlite3"
-    end
+    # ( This pattern can be adapted to additional optional modules.  Just remember to
+    # include a space in front of the module name )
+    module_list << " gpgsql" if build.with? "postgresql"
+    args << module_list
```

Comment:
The previous code didn't allow anybody to add other `--with-<backend>` options down the line, which is why I refactored it.  I'm not changing it back. You can decide whether you want to take it or leave it.